test-data/unit/check-generics.test

```diff
@@ -1505,7 +1505,7 @@ T = TypeVar('T')
 class C(Generic[T]):
     def __init__(self) -> None: pass
 x = C # type: Callable[[], C[int]]
-y = C # type: Callable[[], int] # E: Incompatible types in assignment (expression has type C[T], variable has type Callable[[], int])
+y = C # type: Callable[[], int] # E: Incompatible types in assignment (expression has type Type[C[T]], variable has type Callable[[], int])
```

Comment:
I am also not entirely sure I understand what's wrong with `Type[C[T]]`. Making this work seems to require either bringing erasure to this module or duplicating the formatting of instance types _without_ type parameters.